windowing_library.py

```diff
@@ -0,0 +1,71 @@
+import pickle
+import matplotlib.pyplot as plt
+import numpy as np
+import skimage 
+from scipy import signal
+from scipy.spatial import procrustes
+from skimage.util.shape import view_as_windows
+import statistics
+
+# make it usable for all rows and make one function you call first for phase 1 and 2 then a second main function for phase 3
+
+def create_windows(matrix, rows, window_size, stride, mod_num):
```

Comment:
since matrix is a dict, it would be better to make that clear in the name. I was expecting the input to be an np array.

Personally, I'd prefer to separate the dict->np.array & windowing functionality in two functions. And, since the windowing only needs view_as_windows, you don't need to wrap that in a separate function